lib/record_store/provider/cloudflare.rb

```diff
@@ -82,74 +80,84 @@ def secrets
       end
 
       def build_api_body(record)
-        api_record = {
+        api_body = {
           name: record.fqdn,
           ttl: record.ttl,
           type: record.type,
         }
         case record
         when Record::A, Record::AAAA
-          api_record[:content] = record.rdata_txt
+          api_body[:content] = record.rdata_txt
         when Record::CNAME
-          api_record[:content] = record.rdata_txt
+          api_body[:content] = record.rdata_txt
         when Record::PTR
-          api_record[:content] = record.rdata_txt
+          api_body[:content] = record.rdata_txt
         when Record::MX
-          api_record[:content] = record.exchange
-          api_record[:priority] = record.preference
+          api_body[:content] = record.exchange
+          api_body[:priority] = record.preference
         when Record::TXT, Record::SPF
-          api_record[:content] = record.txtdata.gsub('\;', ';')
+          api_body[:content] = record.txtdata.gsub('\;', ';')
         when Record::CAA
-          api_record[:data] = record.rdata
+          api_body[:data] = record.rdata
         when Record::SRV
-          api_record[:data] = rdata
+          api_body[:data] = rdata
+        when Record::ALIAS
+          api_body[:type] = 'CNAME'
+          api_body[:content] = record.rdata_txt
+          api_body[:settings] = { flatten_cname: true }
+        when Record::NS
+          api_body[:content] = record.nsdname
         end
 
-        api_record
+        api_body
       end
 
-      def build_from_api(api_record)
-        fqdn = Record.ensure_ends_with_dot(api_record['name'])
+      def build_from_api(api_response)
+        fqdn = Record.ensure_ends_with_dot(api_response['name'])
 
-        record_type = api_record['type']
+        record_type = api_response['type']
 
         record = {
-          record_id: api_record['id'],
-          ttl: api_record['ttl'],
+          record_id: api_response['id'],
+          ttl: api_response['ttl'],
           fqdn: fqdn.downcase,
         }
 
         case record_type
         when 'A', 'AAAA'
-          record.merge!(address: api_record['content'])
+          record.merge!(address: api_response['content'])
         when 'CNAME'
-          record.merge!(cname: api_record['content'])
+          if api_response['settings']['flatten_cname'] == false
+            record.merge!(cname: api_response['content'])
+          elsif api_response['settings']['flatten_cname'] == true
+            record_type = 'ALIAS'
+            record.merge!(alias: api_response['content'])
+          end
```

Comment:
```suggestion
          if api_response.dig('settings', 'flatten_cname')
            record_type = 'ALIAS'
            record.merge!(alias: api_response['content'])
          else
            record.merge!(cname: api_response['content'])
          end
```

is `settings` guaranteed to be present?
maybe consider this as an alternative?